### PR TITLE
Add SPDM 1.0 compatibility

### PR DESCRIPTION
**Pull request by jklmnn**
- Allow SPDM 1.0 and add additional messages and refinements where required
 - Fix inconsistencies with spec
 - Add tests

### CODE REVIEW

File: spdm.rflx [hunk 1]
```diff
@@ -254,29 +264,47 @@ package SPDM is
        PSK_Resp_With_Context    => 2)
    with Size => 2;
 
-   type Capabilities_Response is
+   type Capabilities_Response_1_0 is
```
Comment (senier) [inline spdm.rflx:267]: The version 1.1 messages should be a variant of the respective 1.0 messages. This is not easily possible right now as we don't have the version information available in the inner message. The parameterized messages implemented in Componolit/RecordFlux#609 are a potential solution (similar to how we want to handle TCP checksums). Please add an `ISSUE: Componolit/RecordFlux#609` and a brief description so we don't forget to refactor it later to *each* message that could be simplified.

**Review by senier — CHANGES_REQUESTED** [review]

File: spdm.rflx (hunk 1 above)
Comment (jklmnn) [reply spdm.rflx:281]: Fixed in https://github.com/Componolit/RecordFlux-specifications/pull/75/commits/cfa18103274b8fdc2a2f5ff1bb0c73e0e33e0645.

File: spdm.rflx [hunk 2]
```diff
@@ -158,7 +158,8 @@ package SPDM is
          Alpha : Alpha
             --  Assume a released version
             then null
-               if Alpha = 0;
+               if Alpha = 0
```
[the remark this answers is not in the document]
Comment (jklmnn) [reply spdm.rflx]: Fixed in https://github.com/Componolit/RecordFlux-specifications/pull/75/commits/cfa18103274b8fdc2a2f5ff1bb0c73e0e33e0645.

File: spdm.rflx [hunk 3]
```diff
@@ -117,7 +117,7 @@ package SPDM is
                if Major_Version = 1;
          Minor_Version : Minor_Version
             then Code
-               if Minor_Version = 1;
+               if Minor_Version = 1 or Minor_Version = 0;
```
[the remark this answers is not in the document]
Comment (jklmnn) [reply spdm.rflx]: Fixed in https://github.com/Componolit/RecordFlux-specifications/pull/75/commits/cfa18103274b8fdc2a2f5ff1bb0c73e0e33e0645

File: spdm.rflx [hunk 4]
```diff
@@ -104,7 +104,7 @@ package SPDM is
...
```
[the remark this answers is not in the document]
Comment (jklmnn) [reply spdm.rflx]: Fixed in https://github.com/Componolit/RecordFlux-specifications/pull/75/commits/cfa18103274b8fdc2a2f5ff1bb0c73e0e33e0645.

**Comment by jklmnn** [comment]
I additionally added some hacks in https://github.com/Componolit/RecordFlux-specifications/pull/75/commits/cfa18103274b8fdc2a2f5ff1bb0c73e0e33e0645 and fixed some field orders and sizes to support `Negotiate_Algorithms` and `Algorithms`.

**Review by senier — CHANGES_REQUESTED** [review]
@jklmnn A test case still fails in the CI.

**Comment by jklmnn** [comment]
> @jklmnn A test case still fails in the CI.

Should be fixed in https://github.com/Componolit/RecordFlux-specifications/pull/75/commits/a3f63b9f4cc44afd467a98427eddf774ebc58d87.

**Review by senier — APPROVED** [review]

**Pull request merged**